tool/randomizer.lua

```diff
@@ -148,9 +160,6 @@ local function nextlevel()
 end
 
 function randomizer.randomize()
```

Comment:
I think it might be better for you to split off a function which handles the random-sequence generation and a function which handles your set-sequence "generation", both returning their generated sequence. That way you don't mix your logic into the logic of the random sequence too much and reduce the (cyclomatic) complexity of this function (i.e. reduce the number of branches in it).

Right now it's pretty hard to see what you changed and what was just moved around.